packages/resource-deployment/scripts/function-app-create.sh

```diff
@@ -28,133 +34,153 @@ addReplyUrlIfNotExists() {
-
-    echo "Copying config file to function dist folders..."
-    for folderName in $dropFolder/web-api/dist/*-func; do
+copyConfigFileToScriptFolder() {
+    echo "Copying config file to '$packageName' script folder..."
+    for folderName in $dropFolder/$packageName/dist/*-func; do
         if [[ -d $folderName ]]; then
             cp "$dropFolder/resource-deployment/dist/runtime-config/runtime-config.$environment.json" "$folderName/runtime-config.json"
-            echo "  Successfully copied config file to $folderName"
+            echo "  Successfully copied '$environment' config file to $folderName"
         fi
     done
 }
 
-# Set default ARM Function App template files
+installAzureFunctionsCoreToolsOnLinux() {
+    # Install the Azure Functions Core Tools. Refer to https://docs.microsoft.com/en-us/azure/azure-functions/functions-run-local#v2
+
+    # Install the Microsoft package repository GPG key, to validate package integrity
+    curl https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor >microsoft.gpg
+    sudo mv microsoft.gpg /etc/apt/trusted.gpg.d/microsoft.gpg
+
+    # Verify your Ubuntu server is running one of the appropriate versions from the table below. To add the apt source, run
+    sudo sh -c 'echo "deb [arch=amd64] https://packages.microsoft.com/repos/microsoft-ubuntu-$(lsb_release -cs)-prod $(lsb_release -cs) main" > /etc/apt/sources.list.d/dotnetdev.list'
+    sudo apt-get update
+
+    # Install the Core Tools package
+    sudo apt-get install azure-functions-core-tools
+}
+
+publishFunctionAppScripts() {
+    currentDir=$(pwd)
+    # Copy config file to function app deployment folder
+    copyConfigFileToScriptFolder $packageName
+
+    # Change directory to the function app scripts folder
+    cd "${0%/*}/../../../$packageName/dist"
+
+    # Publish the function scripts to the function app
+    echo "Publishing '$packageName' scripts..."
+    func azure functionapp publish $functionAppName --node
+
+    cd "$currentDir"
+}
+
+# Set default ARM Azure Function App template files
 templateFilePath="${0%/*}/../templates/function-app-template.json"
 
 # Read script arguments
-while getopts "r:c:e:k:d:" option; do
+while getopts "r:c:e:k:d:x:g:" option; do
     case $option in
     r) resourceGroupName=${OPTARG} ;;
     c) clientId=${OPTARG} ;;
     e) environment=${OPTARG} ;;
     k) keyVault=${OPTARG} ;;
     d) dropFolder=${OPTARG} ;;
+    x) functionAppNamePrefix=${OPTARG} ;;
+    g) packageName=${OPTARG} ;;
     *) exitWithUsageInfo ;;
     esac
 done
 
-if [ -z $resourceGroupName ] || [ -z $environment ] || [ -z $keyVault ]; then
+if [ -z $resourceGroupName ] || [ -z $environment ] || [ -z $keyVault ] || [ -z $packageName ]; then
     exitWithUsageInfo
 fi
 
-if [ -z $clientId] && [ ! "${environment,,}" = "dev" ]; then
-    echo "Client ID is required if the environment is not Dev"
-    exitWithUsageInfo
+if [ -z $functionAppNamePrefix ]; then
+    functionAppNamePrefix="$packageName-allyfuncapp"
```

Comment:
you are changing the function app name which is is already deployed in production. so, this may create new azure function for the rest api as well?